src/util/db/dbconnection.cpp

```diff
@@ -0,0 +1,418 @@
+#include "util/db/dbconnection.h"
+
+#include <QDir>
+#include <QSqlDriver>
+#include <QSqlError>
+
+#ifdef __SQLITE3__
+#include <sqlite3.h>
+#endif // __SQLITE3__
+
+#include "util/string.h"
+#include "util/assert.h"
+
+
+// Originally from public domain code:
+// http://www.archivum.info/qt-interest@trolltech.com/2008-12/00584/Re-%28Qt-interest%29-Qt-Sqlite-UserDefinedFunction.html
+
+namespace {
+
+const QString kDatabaseType = "QSQLITE";
+
+const QString kDatabaseHostName = "localhost";
+const QString kDatabaseFileName = "mixxxdb.sqlite";
+const QString kDatabaseUserName = "mixxx";
+const QString kDatabasePassword = "mixxx";
+
+void makeLatinLow(QChar* c, int count) {
+    for (int i = 0; i < count; ++i) {
+        if (c[i].decompositionTag() != QChar::NoDecomposition) {
+            c[i] = c[i].decomposition()[0];
+        }
+        if (c[i].isUpper()) {
+            c[i] = c[i].toLower();
+        }
+    }
+}
+
+const QChar LIKE_MATCH_ONE = '_';
```

Comment:
I think this code was originally copied unchanged from the source. 
Now that we make to Mixxx code, the style doe not match. If you have fun, you could finish the restyling ... 

const QChar kLikeMatchOne = '_';